internal/service/loadbalancer/backend_tls_config.go

```diff
@@ -2,153 +2,261 @@ package loadbalancer
-	if err := utils.UnmarshalID(d.Id(), &serviceID, &beName, &name); err != nil {
-		return diag.FromErr(err)
+func (r *backendTLSConfigResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data backendTLSConfigModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
 	}
-	t, err := svc.GetLoadBalancerBackendTLSConfig(ctx, &request.GetLoadBalancerBackendTLSConfigRequest{
-		ServiceUUID: serviceID,
-		BackendName: beName,
-		Name:        name,
-	})
-	if err != nil {
-		return utils.HandleResourceError(d.Get("name").(string), d, err)
+
+	if data.ID.ValueString() == "" {
+		resp.State.RemoveResource(ctx)
+
+		return
 	}
 
-	if err = d.Set("backend", utils.MarshalID(serviceID, beName)); err != nil {
-		return diag.FromErr(err)
+	var loadbalancer, backendName string
+	err := utils.UnmarshalID(data.Backend.ValueString(), &loadbalancer, &backendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer backend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
 
-	if diags = setBackendTLSConfigResourceData(d, t); len(diags) > 0 {
-		return diags
+	tlsConfig, err := r.client.GetLoadBalancerBackendTLSConfig(ctx, &request.GetLoadBalancerBackendTLSConfigRequest{
+		ServiceUUID: loadbalancer,
+		BackendName: backendName,
+		Name:        data.Name.ValueString(),
+	})
+	if err != nil {
+		if utils.IsNotFoundError(err) {
+			resp.State.RemoveResource(ctx)
+		} else {
+			resp.Diagnostics.AddError(
+				"Unable to read loadbalancer backend TLS config details",
+				utils.ErrorDiagnosticDetail(err),
+			)
+		}
+		return
 	}
 
-	return diags
+	resp.Diagnostics.Append(setBackendTLSConfigValues(ctx, &data, tlsConfig)...)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
-func resourceBackendTLSConfigUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) (diags diag.Diagnostics) {
-	svc := meta.(*service.Service)
-	var serviceID, beName, name string
-	if err := utils.UnmarshalID(d.Id(), &serviceID, &beName, &name); err != nil {
-		return diag.FromErr(err)
+func (r *backendTLSConfigResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data backendTLSConfigModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	var loadbalancer, backendName string
+	err := utils.UnmarshalID(data.Backend.ValueString(), &loadbalancer, &backendName)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Unable to unmarshal loadbalancer backend name",
+			utils.ErrorDiagnosticDetail(err),
+		)
+
+		return
 	}
-	t, err := svc.ModifyLoadBalancerBackendTLSConfig(ctx, &request.ModifyLoadBalancerBackendTLSConfigRequest{
-		ServiceUUID: serviceID,
-		BackendName: beName,
-		Name:        name,
+
+	apiReq := &request.ModifyLoadBalancerBackendTLSConfigRequest{
+		ServiceUUID: loadbalancer,
+		BackendName: backendName,
+		Name:        data.Name.ValueString(),
```

Comment:
Should this name come from the ID (or state) as this should be the current name instead of the planned name?

Reply:
See comment below